server/plugin/api.go

```diff
@@ -233,6 +234,37 @@ func (p *Plugin) handleUnlinkProject(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// handleGetUserAccountDetails provides user details
+func (p *Plugin) handleGetUserAccountDetails(w http.ResponseWriter, r *http.Request) {
+	mattermostUserID := r.Header.Get(constants.HeaderMattermostUserIDAPI)
+
+	userDetails, err := p.Store.LoadUser(mattermostUserID)
+	if err != nil {
+		p.API.LogError(constants.ErrorLoadingDataFromKVStore, "Error", err.Error())
+		p.handleError(w, r, &serializers.Error{Code: http.StatusBadRequest, Message: err.Error()})
```

Comment:
```suggestion
		p.handleError(w, r, &serializers.Error{Code: http.StatusInternalServerError, Message: err.Error()})
```
The user ID cannot be wrong because it's being validated by Mattermost server before forwarding the request here so the status code should be 500 here.